geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaDataStoreSchemaManager.scala

```diff
@@ -137,6 +138,38 @@ trait KafkaDataStoreSchemaManager extends DataStore with LazyLogging {
     }
   }
 
+  def updateKafkaSchema(typeName: String, sft: SimpleFeatureType) = {
+
+    // Get previous schema and user data
+    val previousSft = getSchema(typeName)
+    val schemaTypeName = sft.getTypeName()
+
+    // Prevent modifying wrong type if type names don't match
+    if (!schemaTypeName.equals(typeName.toString)) {
+      throw new UnsupportedOperationException("Updating the type name of a schema is not allowed " + schemaTypeName + " " + typeName)
```

Comment:
I'm not sure this is the proper Exception to throw. This gets thrown when the user passes in incorrect args, so maybe an IllegalArgumentException? If the typeName and sft.getTypeName that the user passes in are different, it's more likely the user made an error rather than them trying to update the type name of a schema.